src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,141 @@
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        if (get(manufacturer.getId()).isPresent()) {
+            Manufacturer manufacturerResult = new Manufacturer();
+            String selectManufacturerRequest =
+                    "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false";
+            String updateRequest = "UPDATE manufacturers SET name = ?, country = ? WHERE id = ?";
+            try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement updateManufacturerStatementUpdate =
+                            connection.prepareStatement(updateRequest,
+                                    Statement.RETURN_GENERATED_KEYS);
+                    PreparedStatement selectManufacturerStatementSelect =
+                            connection.prepareStatement(selectManufacturerRequest,
+                                 Statement.RETURN_GENERATED_KEYS)) {
+                updateManufacturerStatementUpdate.setString(1, manufacturer.getName());
+                updateManufacturerStatementUpdate.setString(2, manufacturer.getCountry());
+                updateManufacturerStatementUpdate.setLong(3, manufacturer.getId());
+                updateManufacturerStatementUpdate.executeUpdate();
+
+                selectManufacturerStatementSelect.setLong(1, manufacturer.getId());
+                ResultSet resultSet = selectManufacturerStatementSelect.executeQuery();
+                while (resultSet.next()) {
+                    Long id = resultSet.getObject("id", Long.class);
+                    String name = resultSet.getString("name");
+                    String country = resultSet.getString("country");
+                    manufacturerResult.setId(id);
+                    manufacturerResult.setName(name);
+                    manufacturerResult.setCountry(country);
+                }
+            } catch (SQLException e) {
+                throw new DataProcessingException("Can't update manufacturer from DB", e);
+            }
+            return manufacturerResult;
+        } else {
+            throw new DataProcessingException("Can't update manufacturer from DB");
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(deleteRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setLong(1, id);
+            return createManufacturerStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer from DB", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't delete manufacturer from DB by id " + id, e);
```
Mae error messages more informative